syntax = 'proto3';


message NumberRequest{
  int32 contact_id = 1;
  string label = 2;
  string phone = 3;
  bool is_default = 4;
}
message NumberResponse{
  string label = 1;
  string phone = 2;
  bool is_default = 3;
  string message = 4;
}

message ContactRequest{
  string first_name = 1;
  string last_name = 2;
  string nick_name = 3;

}
message ContactResponse {
  string message = 1;
  string first_name = 2;
  string last_name = 3;
  string nick_name = 4;
  repeated NumberResponse numbers = 5;
  int32 id = 6;
  string created = 7;
}

message ReteriveContactRequest{
  int32 contact_id = 1;
}

service ContactService {
  rpc GetContact(ReteriveContactRequest) returns (ContactResponse){};
  rpc AddContact(ContactRequest) returns (ContactResponse){};
}

service NumberService{
  rpc AddNumber(NumberRequest) returns (NumberResponse){};
}